src/random.cc

```diff
@@ -41,16 +42,19 @@ double rng_t::randdouble() {
     return std::uniform_real_distribution<double>(0, 1)(m_mt19937);
 }
 
-TLS_with_init(rng_t*, rng, nullptr)
+// TLS_with_init(rng_t*, rng, nullptr)
 
 rng_t *get_TLS_rng() {
     // This lazy construction is to ensure that we construct the thread local
     // rng_t on the correct thread, instead of creating it on startup.
 
-    if (TLS_get_rng() == nullptr) {
-        TLS_set_rng(new rng_t());
-    }
-    return TLS_get_rng();
+    // if (TLS_get_rng() == nullptr) {
+    //     TLS_set_rng(new rng_t());
+    // }
+    // return TLS_get_rng();
+    
+    static thread_local std::unique_ptr<rng_t> tls_rng(new rng_t());
```

Comment:
Well.  That looks great.  I'll check that this works on the oldest platforms we support, and if it does, we'll merge this and backport it to v2.4.x.

Thanks for the PR.